system/jlib/jsuperhash.cpp

```diff
@@ -483,26 +483,36 @@ void SuperHashTable::doKill(void)
 
 void SuperHashTable::releaseAll(void)
 {
-    unsigned i;
-    for (i = 0; i < tablesize; i++)
+    if (tablecount)
     {
-        void * et = table[i];
-        table[i] = NULL;
-        if (et)
-            onRemove(et);
+        unsigned i;
+        for (i = 0; i < tablesize; i++)
+        {
+            void * et = table[i];
+            table[i] = NULL;
+            if (et)
+                onRemove(et);
+        }
+        tablecount = 0;
+        setCache(0);
```

Comment:
Not resetting the cache (previous behaviour), meant a potential inconsequential condition in doFindExact being met.
Is that the only consequence?

Trivial, but was slightly faster before when left set (>tablesize) as it meant doFindExact avoided the 2nd expression test (table[i] != et)